name-hash.c

```diff
@@ -23,17 +23,23 @@ static int dir_entry_cmp(const struct dir_entry *e1,
 			name ? name : e2->name, e1->namelen);
 }
 
-static struct dir_entry *find_dir_entry(struct index_state *istate,
-		const char *name, unsigned int namelen)
+static struct dir_entry *find_dir_entry__hash(struct index_state *istate,
```

Comment:
The commit looks overall very reasonable. I will probably change this to have a single `find_dir_entry()` in the end, though, forcing all callers to hash themselves. This will help future callers avoid the same problem that was fixed by the commit.